zerver/lib/bugdown/__init__.py

```diff
@@ -1596,6 +1610,10 @@ def extendMarkdown(self, md: markdown.Markdown, md_globals: Dict[str, Any]) -> N
                              AutoNumberOListPreprocessor(md),
                              "_begin")
 
+        md.preprocessors.add('translate_emoticons',
+                             EmoticonTranslationPreprocessor(md),
+                             '_begin')
```

Comment:
I'm not usre `_begin` is the right place in the ordering for this to go, but it might be.  The way to tell easily is by having tests for potential ordering issues (e.g. putting the syntax inside various types of code blocks; emoticons shouldn't be translated in that context).